apps/console.py

```diff
@@ -462,6 +463,43 @@ async def do_read(self, params):
         else:
             self.show_error('no such characteristic')
 
+    async def do_write(self, params):
+        if not self.connected_peer:
+            self.show_error('not connected')
+            return
+
+        if len(params) != 2:
+            self.show_error('invalid syntax', 'expected write <attribute> <value>')
+            return
+
+        if params[1].upper().startswith("0X"):
+            value = bytes.fromhex(params[1][2:])  # parse as hex string
+        else:
+            try:
+                value = int(params[1])  # try as integer
+            except ValueError:
+                value = str.encode(params[1])  # must be a string
+
+        parts = params[0].split('.')
+        if len(parts) == 2:
+            service_uuid = UUID(parts[0]) if parts[0] != '*' else None
+            characteristic_uuid = UUID(parts[1])
+            for service in self.connected_peer.services:
+                if service_uuid is None or service.uuid == service_uuid:
+                    for characteristic in service.characteristics:
+                        if characteristic.uuid == characteristic_uuid:
+                            await self.connected_peer.write_value(
+                                characteristic,
+                                value,
+                                with_response=(
+                                    characteristic.properties & Characteristic.WRITE
+                                ),  # use write with response, if supported
+                            )
+                            return
+            self.show_error('no such characteristic')
+        else:
+            self.show_error('no such characteristic')
```

Comment:
It would be nice to support specifying the characteristic by its handle, as is done for `read`. Both `read_value` and `write_value` should support passing in either an attribute handle or a characteristic. The code to parse the command and retrieve the handle or characteristic could be factored out and shared between the `read` and `write` commands.

Reply:
Sure thing. Added a `find_attribute()` method to look up a characteristic or attribute handle for `do_read()` and `do_write()`.